kratos/tests/test_numpy_export_dense_matrix.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import print_function, absolute_import, division
+
+import KratosMultiphysics
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+try:
+    import numpy as np
+    missing_numpy = False
+except ImportError as e:
+    missing_numpy = True
+
+class TestNumpyExportDenseMatrix(KratosUnittest.TestCase):
+
+    @KratosUnittest.skipIf(missing_numpy,"Missing python libraries (numpy)")
+    def test_numpy_export_dense_matrix(self):
+        # Create a Kratos matrix
+        KratosMatrix = KratosMultiphysics.Matrix(3,3)
+        KratosMatrix.fill(1.0)
+        
+        # Export it to numpy array (No copying is performed)
+        NumpyMatrix = np.array(KratosMatrix,copy=False)
+
+        # Test correct creation
+        for i in range(3):
+            for j in range(3):
+                self.assertAlmostEqual(KratosMatrix[i,j], NumpyMatrix[i,j], 1e-12)
+        
+        #Change an entry in Kratos Matrix
+        KratosMatrix[0,0] = 123
+
+        #Test change in Numpy matrix  
+        self.assertAlmostEqual(KratosMatrix[0,0], NumpyMatrix[0,0], 1e-12)
+
+        #Change an entry in Numpy matrix
+        NumpyMatrix[2,2] = 555
+        
+        #Test change in Kratos matrix    
+        self.assertAlmostEqual(KratosMatrix[2,2], NumpyMatrix[2,2], 1e-12)
```

Comment:
Should be Equal?

Reply:
no, its double

Reply:
@loumalouomega I think you are right and it should be AssertEqual, since both are pointing to the same address. In the case of the copy, I kept on using assertAlmostEqual...